src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -262,7 +267,11 @@ private bool MatchString(string str)
 
             if (!_rightToLeft)
             {
-                pos += str.Length;
+                //pos += str.Length;
```

Comment:
?

Reply:
To compensate for the change above. Previously we were decreasing `pos` by `str.Length` while looking for a match => the next position to process is at `pos + str.Length`. Now that I'm using `SequenceEqual`, `pos` doesn't need to change.